templates/EFS.yml

```diff
@@ -70,6 +88,162 @@ Resources:
 
 {%   endfor %}
 
+{%  if filesystem.s3_to_efs_copy is defined %}
+# Lambda to EFS copy
+  S3ToEfsBucket{{ filesystem.cfn_name }}:
+    Type: AWS::S3::Bucket
+    DependsOn:
+      - AwsLambdaS3ToEfsCopy{{ filesystem.cfn_name }}InvokePermission
+    Properties:
+      AccessControl: "Private"
+      BucketName: !Sub "{{ application }}-{{ env }}-efs-${ {{ filesystem.cfn_name }} }"
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: AES256
+      VersioningConfiguration:
+        Status: Suspended
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: "true"
+        BlockPublicPolicy: "true"
+        IgnorePublicAcls: "true"
+        RestrictPublicBuckets: "true"
+      CorsConfiguration:
+        CorsRules:
+          - AllowedHeaders:
+              - '*'
+            AllowedMethods:
+              - 'GET'
+              - 'PUT'
+            AllowedOrigins:
+              - '*'
+      LifecycleConfiguration:
+        Rules:
+        - NoncurrentVersionExpirationInDays: 30
+          ExpirationInDays: 30
+          Status: Enabled
+      NotificationConfiguration:
+        LambdaConfigurations:
+          - Event: s3:ObjectCreated:*
+            Function: !GetAtt AwsLambdaS3ToEfsCopy{{ filesystem.cfn_name }}.Arn
+            Filter:
+              S3Key:
+                Rules:
+                  - Name: suffix
+                    Value: ".json"
+                  - Name: prefix
+                    Value: "trigger"
+      Tags:
+        - Key: Application
+          Value: "{{ application }}"
+        - Key: Environment
+          Value: "{{ env }}"
+        - Key: Customer
+          Value: "{{ customer | default('NA') }}"
+
+  AwsLambdaS3ToEfsCopy{{ filesystem.cfn_name }}:
+    Type: AWS::Lambda::Function
+    DependsOn:
+      - {{ filesystem.cfn_name }}MountTargetPrivateSubnetAZ1
+      - {{ filesystem.cfn_name }}MountTargetPrivateSubnetAZ2
+      - {{ filesystem.cfn_name }}MountTargetPrivateSubnetAZ3
+    Properties:
+      Handler: "aws-lambda-s3-to-efs-copy.lambda_handler"
+      Role: !GetAtt LambdaEfsRole{{ filesystem.cfn_name }}.Arn
+      Code:
+        S3Bucket: "{{ lambda_function_bucket_name }}"
+        S3Key: "{{ filesystem.s3_to_efs_copy.lambda_key }}"
+      Runtime: python3.8
+      Environment:
+        Variables:
+          PROJECT: "{{ application }}"
+          ENVIRONMENT: "{{ env }}"
+          FILESYSTEM: "{{ filesystem.cfn_name }}"
```

Comment:
Does the function really need/use this envvar?

Reply:
These variables are used to send the slack notification